### ISSUE TITLE
Merging of trajectories with Brian 2 parameters/results fails - AttributeError: 'tuple' object has no attribute 'extend'

### ISSUE THREAD

**Issue by Felix11H** [SmokinCaterpillar/pypet#50]
Brian2==2.0.2.1, pypet==0.4.1

First generate trajectories

```python
from pypet import cartesian_product
from pypet.environment import Environment
from pypet.brian2.parameter import Brian2Parameter, Brian2MonitorResult

from brian2.units import ms,mV
from brian2 import NeuronGroup, StateMonitor, run


def run_net(tr):
    namespace = tr.netw.f_to_dict(short_names=True, fast_access=True)
    GExc = NeuronGroup(N=1, model='dV/dt=-V/tau+sigma*xi/(tau**0.5): volt',
                       namespace=namespace)
    GExc_stat = StateMonitor(GExc, ['V'], record=[0])
    run(tr.sim_t)
    tr.v_standard_result = Brian2MonitorResult
    tr.f_add_result('GExc_stat', GExc_stat)

    
filename = 'envd.hdf5'

# -------------------  trajectory 1 ----------------------

env = Environment(trajectory='tr1',
                  add_time=False,
                  filename=filename,
                  overwrite_file=False)

tr = env.trajectory

tr.v_standard_parameter=Brian2Parameter
tr.v_fast_access=True
tr.f_add_parameter('netw.tau', 20*ms)
tr.f_add_parameter('netw.sim_t', 100*ms)
tr.f_add_parameter('netw.sigma', 5.0**0.5*mV)

tr.f_explore(cartesian_product({'sim_t':[100,200]*ms,
                                'sigma':[5.0,6.0]*mV}))

env.run(run_net)


# -------------------  trajectory 2 ----------------------

env = Environment(trajectory='tr2',
                  add_time=False,
                  filename=filename,
                  overwrite_file=False)

tr = env.trajectory

tr.v_standard_parameter=Brian2Parameter
tr.v_fast_access=True
tr.f_add_parameter('netw.tau', 20*ms)
tr.f_add_parameter('netw.sim_t', 100*ms)
tr.f_add_parameter('netw.sigma', 5.0**0.5*mV)

tr.f_explore(cartesian_product({'sim_t':[300,400]*ms,
                                'sigma':[5.0,6.0]*mV}))

env.run(run_net)
```

Try to merge

```python

from brian2.units import mV, ms, second

from pypet.trajectory import Trajectory
from pypet.brian2.parameter import Brian2Parameter, Brian2MonitorResult

tr1 = Trajectory(name='tr1',
                  add_time=False,
                  filename='envd.hdf5',
                  dynamic_imports=[Brian2MonitorResult, Brian2Parameter])

tr1.f_load(load_parameters=2, load_derived_parameters=2, load_results=2)


tr2 = Trajectory(name='tr2',
                  add_time=False,
                  filename='envd.hdf5',
                  dynamic_imports=[Brian2MonitorResult, Brian2Parameter])

tr2.f_load(load_parameters=2, load_derived_parameters=2, load_results=2)

# returns an error
tr1.f_merge(tr2, backup_filename=True, move_data=True, 
                      delete_other_trajectory = True)
```

I get 

`AttributeError: 'tuple' object has no attribute 'extend'`

**Comment by SmokinCaterpillar** [comment]
Sorry, for the delay, but I am looking into it right now :-)

**Comment by SmokinCaterpillar** [comment]
Definitely a bug. The problem can be found inthe `brian2/parameter.py` file:

```
def _load(self, load_dict):
        if self.v_locked:
            raise pex.ParameterLockedException('Parameter `%s` is locked!' % self.v_full_name)

        try:
            data_table = load_dict['data' + Brian2Parameter.IDENTIFIER]

            unitstring = data_table['unit'][0]
            unit = unit_from_expression(unitstring)
            value = data_table['value'][0]
            self._data = value * unit

            if 'explored_data' + Brian2Parameter.IDENTIFIER in load_dict:
                explore_table = load_dict['explored_data' + Brian2Parameter.IDENTIFIER]

                value_col = explore_table['value']
                explore_list = [value * unit for value in value_col]

                ################ The Error stems from here -.- #####################
                self._explored_range = tuple(explore_list)
                ....
```

**Comment by SmokinCaterpillar** [comment]
Can you check if replacing 
`self._explored_range = tuple(explore_list)` simply by
`self._explored_range = explore_list` solves your problem? Thanks!

**Comment by Felix11H** [comment]
Thanks much for looking into this. I applied the change and it seems the merge in the example above was successful, all expected runs are present in the merged trajectory.

When merging with the exact script from above, I get an ouput:
```
Your loading was not successful, could not find a single item to load.
Your loading was not successful, could not find a single item to load.
```
but script completes without returning errors. Is this printed output expected/intended/helpful?

**Comment by SmokinCaterpillar** [comment]
Hi, no this is output is indeed misleading. The trajectory tries to load derived parameters but doesn't find any in the process of merging. The warning will be removed in the next release. Thanks for your help and feedback!

**Comment by SmokinCaterpillar** [comment]
Btw, I wrote a regression test for the bug you found and fixed it as suggested above. I will try to update a new release soon once all tests and CI pipelines pass. Maybe I'll upload the new version to pypi tomorrow or the day after, but most definitely till the end of this week.

**Comment by Felix11H** [comment]
Excellent! Thank you for your continued support of pypet!

**Comment by SmokinCaterpillar** [comment]
Fixed in 0.4.2
